test-data/unit/check-literal.test

```diff
@@ -0,0 +1,644 @@
+--
+-- Check to see how we handle raw types, error handling, and other
+-- semantic analysis shenanigans
+--
+
+[case testLiteralInvalidString]
+from typing_extensions import Literal
+def f1(x: 'A[') -> None: pass  # E: Invalid type: syntax error in type hint
+def g1(x: Literal['A[']) -> None: pass
+reveal_type(f1)  # E: Revealed type is 'def (x: Any)'
+reveal_type(g1)  # E: Revealed type is 'def (x: Literal['A['])'
+
+def f2(x: 'A B') -> None: pass  # E: Invalid type: syntax error in type hint
+def g2(x: Literal['A B']) -> None: pass
+reveal_type(f2)  # E: Revealed type is 'def (x: Any)'
```

Comment:
I would also check that `Literal[int]`, `Literal[3.1415]`, and `Literal[Any]` also turn into `Any` (just in case).

Reply:
We already had a test checking the second one, but I added/tweaked test cases to test the first and third ones.